google/cloud/aiplatform/jobs.py

```diff
@@ -1170,9 +1188,18 @@ def run(
                 restart_job_on_worker_restart=restart_job_on_worker_restart,
             )
 
+        if tensorboard:
+            v1beta1_gca_resource = gca_custom_job_v1beta1.CustomJob()
+            v1beta1_gca_resource._pb.MergeFromString(
```

Comment:
I assume that `gca_custom_job_v1beta1` is needed because a certain api_client version is needed and not because Tensorboard is needed?

This code `if tensorboard:` makes it seem like they are explicitly coupled.

Reply:
Yes I understand. Just saying that ideally the `if statement` is split in two parts to decouple logic in case other features besides tensorboard also need `v1beta1`.

if is_v1beta1_needed:            
    v1beta1_gca_resource = gca_custom_job_v1beta1.CustomJob()            
    v1beta1_gca_resource._pb.MergeFromString(self._gca_resource._pb.SerializeToString())            
    self._gca_resource = v1beta1_gca_resource            

if tensorboard:
    self._gca_resource.job_spec.tensorboard = tensorboard

Reply:
Not blocking, but just something to think about.

Reply:
Ack. I'll leave as is and when we introduce a new v1beta1 feature we'll refactor. Thanks!